ci/supervisor/config/config_test.go

```diff
@@ -0,0 +1,472 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// errorMismatch compares two errors by their error string
+func errorMismatch(got error, want error) string {
+	exp := "<no error>"
+	act := "<no error>"
+	if want != nil {
+		exp = want.Error()
+	}
+	if got != nil {
+		act = got.Error()
+	}
+	if exp == act {
+		return ""
+	}
+	return fmt.Sprintf("got: '%v'\nwant: '%v'", act, exp)
+}
+
+func TestParseRepoConfig(t *testing.T) {
+	datas := []struct {
+		yaml string
+		exp  RepoConfig
+		err  error
+	}{
+		{ // empty YAML
+			"",
+			RepoConfig{
+				PresubmitTasks:  map[string]PresubmitTask{},
+				PeriodicTasks:   map[string]PeriodicTask{},
+				PostsubmitTasks: map[string]PostsubmitTask{},
+			},
+			nil,
+		}, { // bad YAML
+			"<>",
+			RepoConfig{},
+			fmt.Errorf("yaml: unmarshal errors:\n  line 1: cannot unmarshal !!str `<>` into map[string]interface {}"),
+		}, { // unknown defaults section
+			`defaults:
+  foobar:
+    optional: true`,
+			RepoConfig{
+				Defaults: Defaults{
+					Settings:     DefaultSettings{},
+					ProwPlugins:  map[string]ProwPlugin{},
+					CodeCoverage: CodeCoverageSettings{},
+				},
+				PresubmitTasks:  map[string]PresubmitTask{},
+				PeriodicTasks:   map[string]PeriodicTask{},
+				PostsubmitTasks: map[string]PostsubmitTask{},
+			},
+			nil,
+		}, { // bad plugin setting
+			`defaults:
+  prow-plugins:
+    plugin1: 0.0`,
+			RepoConfig{},
+			fmt.Errorf("valid setting for plugin \"plugin1\" is one of [disabled|no|false|<configuration>]"),
+		}, { // bad plugin string setting
+			`defaults:
+  prow-plugins:
+    plugin1: foo`,
+			RepoConfig{},
+			fmt.Errorf("valid setting for plugin \"plugin1\" is one of [disabled|no|false|<configuration>]"),
+		}, { // presubmit task definition with unrecognized key
+			`presubmit-tasks:
+  foo-tests:
+    thisisanunrezognizedkey: "foo"`,
+			RepoConfig{
+				PresubmitTasks:  map[string]PresubmitTask{"foo-tests": {}},
+				PeriodicTasks:   map[string]PeriodicTask{},
+				PostsubmitTasks: map[string]PostsubmitTask{},
+			},
+			nil,
+		}, { // periodic task definition with unrecognized key
+			`periodic-tasks:
+  foo-tests:
+    thisisanunrezognizedkey: "foo"`,
+			RepoConfig{
+				PresubmitTasks:  map[string]PresubmitTask{},
+				PeriodicTasks:   map[string]PeriodicTask{"foo-tests": {}},
+				PostsubmitTasks: map[string]PostsubmitTask{},
+			},
+			nil,
+		}, { // postsubmit task definition with unrecognized key
+			`postsubmit-tasks:
+  foo-tests:
+    thisisanunrezognizedkey: "foo"`,
+			RepoConfig{
+				PresubmitTasks:  map[string]PresubmitTask{},
+				PeriodicTasks:   map[string]PeriodicTask{},
+				PostsubmitTasks: map[string]PostsubmitTask{"foo-tests": {}},
+			},
+			nil,
+		}, { // full YAML
+			`defaults:
```

Comment:
might be my personal preference, I feel it better readable having this kind of test data in separated files, so that dev's favorite editor can have proper syntax highlights for yaml. I have seen an example here:
https://github.com/jstemmer/go-junit-report/tree/master/testdata